### PR TITLE
Fix issue 726: Missing selector in Broker resource for selecting security groups 

### PR DESCRIPTION
**Pull request by svscheg**
### Description of your changes
Missing selector in Broker resource for selecting security groups: add selectors for the securityGroup

Now can use next selectors:
securityGroupRefs
securityGroupSelector

Fixes https://github.com/upbound/provider-aws/issues/726

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually
Uptests: https://github.com/upbound/provider-aws/actions/runs/5507977809

### CODE REVIEW

**Comment by svscheg** [comment]
/test-examples="examples/mq/broker.yaml"

File: examples-generated/mq/broker.yaml [hunk 1]
```diff
@@ -18,8 +18,8 @@ spec:
     engineVersion: 5.15.9
     hostInstanceType: mq.t2.micro
     region: us-west-1
-    securityGroups:
-    - ${aws_security_group.test.id}
+    securityGroupRefs:
+    - name: test
```
Comment (turkenf) [inline examples-generated/mq/broker.yaml:22]: Could you please add this parameter and related resources to `examples/mq/broker.yaml` and test it?

**Comment by svscheg** [comment]
/test-examples="examples/mq/broker.yaml"

File: examples-generated/mq/broker.yaml (hunk 1 above)
Comment (svscheg) [reply examples-generated/mq/broker.yaml:22]: Done.
Also added testhooks for this resource.
Also manually tested case when we need to use next selector:
`securityGroupSelector:
  matchLabels:
    testing.upbound.io/example-name: example`

**Review by turkenf — APPROVED** [review]
Thank you @svscheg, lgtm.

**Pull request merged**